myst_nbs/case_studies/reinforcement_learning.myst.md

```diff
@@ -0,0 +1,540 @@
+    logp_actions = logp_actions[at.arange(actions.shape[0] - 1), actions[1:]]
+    return at.sum(logp_actions)  # PyMC expects the standard log-likelihood
+```
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+  height: 245
+id: XQNBZLMvAdbo
+outputId: 65d7a861-476c-4598-985c-e0b0fcd744c4
+---
+with pm.Model() as m:
+    alpha = pm.Beta(name="alpha", alpha=1, beta=1)
+    beta = pm.HalfNormal(name="beta", sigma=10)
+
+    like = pm.Potential(name="like", var=aesara_llik_td(alpha, beta, actions, rewards))
+
+    tr = pm.sample(random_seed=rng)
+```
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+  height: 539
+id: vgSumt-oATfN
+outputId: eb3348a4-3092-48c8-d8b4-678af0173079
+---
+az.plot_trace(data=tr);
+```
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+  height: 408
+id: BL84iT_RAzEL
+outputId: dcd4174b-4148-45cb-f72d-973f1487d8c2
+---
+az.plot_posterior(data=tr, ref_val=[true_alpha, true_beta]);
+```
+
++++ {"id": "1FtAp76PBLCr"}
+
+In this example, the obtained posteriors are nicely centered around the MLE values. What we have gained is an idea of the plausible uncertainty around these values.
+
+### Bonus: Alternative model using Bernoulli for the likelihood
+
+```{code-cell} ipython3
+:id: pQdszDk_qYCX
+
+def right_action_probs(alpha, beta, actions, rewards):
+    rewards = at.as_tensor_variable(rewards, dtype="int32")
+    actions = at.as_tensor_variable(actions, dtype="int32")
+
+    # Compute the Qs values
+    Qs = 0.5 * at.ones((2,), dtype="float64")
+    Qs, updates = aesara.scan(
+        fn=update_Q, sequences=[actions, rewards], outputs_info=[Qs], non_sequences=[alpha]
+    )
+
+    # Apply the sotfmax transformation
+    Qs = Qs[:-1] * beta
+    logp_actions = Qs - at.logsumexp(Qs, axis=1, keepdims=True)
+
+    # Return the probabilities for the right action, in the original scale
+    return at.exp(logp_actions[:, 1])
+```
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+  height: 121
+id: S55HgqZiTfpa
+outputId: a2db2d68-8bf3-4773-8368-5b6dff310e4b
+---
+with pm.Model() as m_alt:
+    alpha = pm.Beta(name="alpha", alpha=1, beta=1)
+    beta = pm.HalfNormal(name="beta", sigma=10)
+
+    action_probs = right_action_probs(alpha, beta, actions, rewards)
+    like = pm.Bernoulli(name="like", p=action_probs, observed=actions[1:])
+
+    tr_alt = pm.sample(random_seed=rng)
+```
+
+```{code-cell} ipython3
+---
+colab:
+  base_uri: https://localhost:8080/
+  height: 452
+id: zjXW103JiDRQ
+outputId: aafc1b1e-082e-414b-cac7-0ad805097057
+---
+az.plot_trace(data=tr_alt);
+```
+
+```{code-cell} ipython3
+:id: SDJN2w117eox
+
+az.plot_posterior(data=tr_alt, ref_val=[true_alpha, true_beta]);
+```
+
+## Watermark
+
+```{code-cell} ipython3
+%load_ext watermark
+%watermark -n -u -v -iv -w -p aeppl,xarray
+```
+
+## References
+
+:::{bibliography}
+:filter: docname in docnames
+:::
+
++++
+
+## Credits
```

Comment:
All the credits/authorship info should be here, not divided in two different sections in the top/bottom of the notebook respectively

Reply:
Thanks! Fixed in https://github.com/pymc-devs/pymc-examples/pull/410/commits/49b3d6b503726f242dc529e7cb3d2a15c0d2bc55

Reply:
The section was not redundant and it should all be placed here. Some deviations are ok but notebooks should follow the jupyter style guide as closely as possible. This includes: https://docs.pymc.io/en/latest/contributing/jupyter_style.html#authorship-and-attribution

Reply:
Ups! Sorry, I was following one of the recent notebooks. I will bring it bak again 🙇

Reply:
Ok, here are all the credits together https://github.com/pymc-devs/pymc-examples/pull/410/commits/61e788123690b4b205c960c819a92b126dfa17e8